### PR TITLE
core/wire: get rid of constant max message size

### PR DESCRIPTION
**Pull request by david415**
Here we instead base our max message bounds on the max padding size for a given command set.
Fixes https://github.com/katzenpost/katzenpost/issues/788

### CODE REVIEW

File: authority/voting/client/client.go [hunk 1]
```diff
@@ -246,7 +246,11 @@ func (p *connector) fetchConsensus(auth *config.Authority, ctx context.Context,
 		return nil, err
 	}
 	p.log.Debugf("sending getConsensus to %s", auth.Identifier)
-	cmd := &commands.GetConsensus{Epoch: epoch}
+	cmd := &commands.GetConsensus{
+		Epoch:              epoch,
+		Cmds:               commands.NewPKICommands(p.cfg.PKISignatureScheme),
+		MixnetTransmission: false, // Disable padding for direct dirauth transmission
```
Comment (ioerror) [inline authority/voting/client/client.go:252]: Is there a security analysis (read: any at all) that says that we want a special case for padding? Wouldn't it just be safer to chunk it up with padding (if over some limit) or pad it (if under that limit)? Having this be distinguishable seems risky - I think the result is that padding is disabled which implies a distinguisher. Is that correct? Can a third party adversary tell that a client is doing a fetch of a specific thing here?

File: bench/bench_test.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ var (
 	)
 )
 
-var testSignatureScheme = signSchemes.ByName("Ed448-Dilithium3")
+var testSignatureScheme = signSchemes.ByName("ed25519")
```
Comment (ioerror) [inline bench/bench_test.go:100]: As a wise person once said: ¿Por qué no los dos?

File: core/wire/commands/old_client_commands.go [hunk 3]
```diff
@@ -138,7 +162,11 @@ func (c *MessageEmpty) ToBytes() []byte {
 	binary.BigEndian.PutUint32(out[2:6], uint32(c.Cmds.messageEmptyLength()))
 	out[6] = byte(messageTypeEmpty)
 	binary.BigEndian.PutUint32(out[8:12], c.Sequence)
-	return padToMaxCommandSize(out, c.Cmds.maxMessageLen(c))
+	return c.Cmds.padToMaxCommandSize(out, false)
+}
+
+func (c *MessageEmpty) Length() int {
+	return cmdOverhead + 4 + c.Cmds.geo.PacketLength
```
Comment (ioerror) [inline core/wire/commands/old_client_commands.go:169]: Why is 4 special and if it is special, can it be given a constant variable name that ensures no one asks this question again?

**Review by ioerror — CHANGES_REQUESTED** [review]

File: bench/bench_test.go (hunk 2 above)
Comment (david415) [reply bench/bench_test.go:100]: it just needs to match whatever signature scheme we've selected by default with genconfig and used for the docker mixnet.

File: authority/voting/client/client.go (hunk 1 above)
Comment (david415) [reply authority/voting/client/client.go:252]: We cannot pad interactions with the PKI. Easier to talk on the phone to explain

File: core/wire/commands/old_client_commands.go (hunk 3 above)
Comment (david415) [reply core/wire/commands/old_client_commands.go:169]: 4 bytes in a uint32

File: bench/bench_test.go (hunk 2 above)
Comment (ioerror) [reply bench/bench_test.go:100]: Okay, I'd make it hybird pq by default.

File: core/wire/commands/old_client_commands.go (hunk 3 above)
Comment (ioerror) [reply core/wire/commands/old_client_commands.go:169]: Okay, understood.

**Review by ioerror — APPROVED** [review]

**Review by ioerror — APPROVED** [review]

**Pull request merged**